ARCHITECTURE.md

```diff
@@ -0,0 +1,32 @@
+# Architecture
+
+This document describes the high-level architecture of realtime_signs. The goal of this document is to provide a useful
+orientation to a new contributor to the codebase.
+
+## Application description
+
+Realtime Signs is the app that puts subway countdown predictions on the black and yellow Daktronics LED signs in and around stations. It does this by continually (about 1/sec) downloading GTFS-rt data and sending HTTP POST requests as necessary to the ARINC head end server.
+
+## Process-level architecture
+
+The smallest granularity with which we can address signs in a station is called the "zone". Each station can have up to six zones: north, south, east, west, center, and mezzanine. Each zone throughout the system has its own dedicated Elixir process, a `Signs.Realtime` GenServer configured at start-up via `priv/signs.json`. See `Signs.Utilities.SourceConfig` for documentation of the format of the `signs.json` file.
```

Comment:
> Each station can have up to six zones: north, south, east, west, center, and mezzanine.

I'd maybe clarify this a bit, though I suppose this knowledge isn't necessary to understand the app:

> Each station can have up to six zones, which are always named "north", "south", "east", "west", "center", and "mezzanine", though these are arbitrary identifiers and may not reflect the real layout of the station.